@here/harp-mapview/lib/text/TextElementsRenderer.ts

```diff
@@ -1709,8 +1727,14 @@ export class TextElementsRenderer {
         }
         // ... and render the icon (if any).
         if (iconReady) {
-            if (iconRejected) {
-                iconRenderState!.startFadeOut(renderParams.time);
+            if (iconRejected || iconInvisible) {
+                if (iconRenderState.isReallyVisible()) {
+                    iconRenderState.startFadeOut(renderParams.time);
```

Comment:
I already commented above that startFadeOut should internally have all checks needed avoid doing anything when it's not needed (already fading out, faded out or undefined labels).

Reply:
I think that is because we're not check for undefined in startFadeOut(). If the label is in undefined state, startFadeOut() should return early like for FadedOut and FadingOut states.